File: docs/src/testers/scraping/pydantic_model.py
# @sniptest filename=pydantic_model.py
# @sniptest show=3-18

from notte_sdk import NotteClient
from pydantic import BaseModel


class Product(BaseModel):
    name: str
    price: float
    description: str


client = NotteClient()
product = client.scrape(
    "https://example.com/product", response_format=Product, instructions="Extract the product details"
)

print(f"Name: {product.name}, Price: {product.price}")

File: docs/src/testers/scraping/pydantic_model.ts
import { z } from 'zod';
import { NotteClient } from 'notte-sdk';

const Product = z.object({
  name: z.string(),
  price: z.number(),
  description: z.string(),
});

const client = new NotteClient({
  apiKey: process.env.NOTTE_API_KEY,
});

const product = await client.scrape('https://example.com/product', {
  response_format: Product,
  instructions: 'Extract the product details',
});

console.log(`Name: ${product.name}, Price: ${product.price}`);
